### PR TITLE
Added Fizzbuzzer class and spec

### PR DESCRIPTION
**Pull request by cpalmerbright**

### CODE REVIEW

File: models/fizzbuzzer.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Fizzbuzzer
+
+  attr_accessor :range
+  attr_writer :fizzer, :buzzer
+
+  def initialize(count)
+    @range = (1..count)
+  end
+
+  def run
```
Comment (saria-b) [inline models/fizzbuzzer.rb]: Awesome work, ,maybe define the fizzes and buzzes before you do the count

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lillypiri) [inline .idea/vcs.xml]: Great work, Caitlin! You can put a gitignore in for the RubyMine stuff.

File: models/fizzbuzzer.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          puts 'Fizz'
+        when buzz
+          puts 'Buzz'
+        else
+          puts number
+      end
+    end
+  end
+
+  def fizzer
```
Comment (nigelr) [inline models/fizzbuzzer.rb]: probably a bit too much extraction

File: specs/models/fizzbuzzer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  it 'play area' do
+    fuzz = Fizzbuzzer.new(100)
+    fuzz.fizzer = 4
+    fuzz.buzzer = 7
+    fuzz.run
+  end
+end
+
+
```
Comment (nigelr) [inline specs/models/fizzbuzzer_spec.rb]: careful with extra lines

File: .idea/vcs.xml (hunk 2 above)
Comment (nigelr) [inline .idea/vcs.xml]: don't include these files

File: models/fizzbuzzer.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
+class Fizzbuzzer
+
+  attr_accessor :range
+  attr_writer :fizzer, :buzzer
+
+  def initialize(count)
+    @range = (1..count)
```
Comment (nigelr) [inline models/fizzbuzzer.rb]: instead try making this class just the extraction and put the iterator external

File: models/fizzbuzzer.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def run
+    @range.each do |number|
+      fizz = fizzes?(number)
+      buzz = buzzes?(number)
+      case
+        when fizz && buzz
+          puts 'Fizzbuzz!'
```
Comment (nigelr) [inline models/fizzbuzzer.rb]: dont do the `puts` here.. return the value.. It makes it easier to test
do the puts in the iterator

File: specs/models/fizzbuzzer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,89 @@
+require 'rspec'
+require_relative "../../models/fizzbuzzer"
+
+describe Fizzbuzzer do
```
Comment (nigelr) [inline specs/models/fizzbuzzer_spec.rb:4]: You should be able to rethink this whole spec suit when the class only checks the value and does not iterate

**Review by nigelr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**